token/program-2022/src/error.rs

```diff
@@ -85,6 +85,9 @@ pub enum TokenError {
     /// Extension already initialized on this account
     #[error("Extension already initialized on this account")]
     ExtensionAlreadyInitialized,
+    /// An account can only be closed if its confidential balance is zero"
+    #[error("An account can only be closed if its confidential balance is zero")]
+    ConfidentialTransferStateHasBalance,
```

Comment:
I don't love how adding extension-specific errors require reaching into `src/error.rs` and `src/processor.rs` but 🙈.